webapp/src/main/java/life/qbic/views/login/LoginHandler.java

```diff
@@ -98,20 +110,19 @@ public void handle(BeforeEvent beforeEvent) {
       String userId = queryParams.get(emailConfirmationParameter).iterator().next();
       confirmEmailInput.confirmEmailAddress(userId);
     }
-    if (queryParams.containsKey("userRegistered")){
-     showEmailConfirmationReminder();
+    if (queryParams.containsKey("userRegistered")) {
+      showEmailConfirmationReminder();
     }
   }
 
   @Override
   public void onEmailConfirmationSuccess() {
-    clearMessages();
+    clearNotifications();
     showEmailConfirmationInformation();
-
   }
 
   @Override
   public void onEmailConfirmationFailure(String reason) {
-    registeredLoginView.showError(new ErrorMessage("Email confirmation failed", reason));
+    showError(new ErrorMessage("Email confirmation failed", reason));
```

Comment:
However I'd argue that since the errormessage is conveying domain logic I'd keep it being created in the domain logic since the methods intention is to show the error message

Reply:
is it really domain logic? its the same class which is part of the infrastucture/application logic

Reply:
You get the text from there that is sufficent I think